roomserver/internal/input_membership.go

```diff
@@ -100,9 +100,9 @@ func updateMembership(
 			return nil, err
 		}
 	}
-	if oldMembership == newMembership && newMembership != gomatrixserverlib.Join {
+	if oldMembership == newMembership || newMembership == gomatrixserverlib.Leave {
 		// If the membership is the same then nothing changed and we can return
-		// immediately, unless it's a Join update (e.g. profile update).
+		// immediately. Likewise if it's a leave - there's nothing more to do.
```

Comment:
Doesn't make sense, because for genuine transitions to `leave` we are returning early - there should be SOMETHING to do, right?